src/osd/OSDMap.cc

```diff
@@ -7330,6 +7333,20 @@ void OSDMap::check_health(CephContext *cct,
       ss << down_in_osds.size() << " osds down";
       auto& d = checks->add("OSD_DOWN", HEALTH_WARN, ss.str(),
 			    down_in_osds.size());
+      bool crimson_build = false;
+      bool crimson_pool_is_configured = false;
+      #ifdef WITH_SEASTAR
+      bool crimson_build = true;
+      bool crimson_pool_is_configured = cmd_getval_or<bool>(cmdmap, "crimson", false) ||
+               cct->_conf->get_val<bool>("osd_pool_default_crimson");
+      #endif
+      if (crimson_build && !crimson_pool_is_configured) {
```

Comment:
I'm not really fond of adding crimson specific build behavior to OSDMap beyond the smart pointer ifdef in the header.  It's *very* important that most of the methods in OSDMap behave identically across the mons, osds, and clients so we don't really want build time behavior changes.

crimson_build doesn't seem to be declared if !WITH_SEASTAR, so it probably won't compile?

Finally, this method will actually be invoked on the monitor, not the crimson-osd, so WITH_SEASTAR will be false regardless.

Reply:
Also, I think it's the allow_crimson flag on the OSDMap that we care about, not the osd_pool_default_crimson config flag.

I don't think you can actually emit the warning from the OSDMap.  If OSDMap::allow_crimson (OSDMap::get_allow_crimson) is false, we simply ignore the boot message from the OSD and its metadata doesn't get added to the map.  See OSDMonitor::preprocess_boot.

Reply:
Thanks for all of this, it's going to help me a lot with understanding ceph in general-I used this PR as a way to learn more about C++, ceph and crimson and the whole development cycle, and comments like this are a huge help so thanks again

Reply:
Yep, no worries.  Ideally, we'd just trigger the health warning from preprocess_boot, but I don't actually remember how the health warning mechanism works.  Feel free to ping me if you can't find a good way to do it and I'll take a look.